### PR TITLE
add carousel to the product details page

### PR DESCRIPTION
**Pull request by jatinbehlEP**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
     "smart-mixin": "^2.0.0",
     "source-map-support": "0.4.0",
     "tether": "^1.4.4",
+    "uglify-js": "^3.4.4",
```
Comment (shaunmaharaj) [inline package.json]: why'd you add this back in?

File: package.json [hunk 2]
```diff
@@ -7,12 +7,12 @@
...
     "babel-runtime": "6.6.1",
     "bluebird": "3.3.3",
-    "body-parser": "1.15.0",
+    "body-parser": "^1.18.3",
     "classnames": "2.2.3",
     "cookie-parser": "1.4.1",
     "dispatchr": "^1.0.0",
     "eventemitter3": "^1.2.0",
-    "express": "4.13.4",
+    "express": "^4.16.3",
```
Comment (shaunmaharaj) [inline package.json]: are you updating these dependencies on purpose?

File: src/ep.config.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "showInstrumentation":false
   },
   "cortexApi":{
-    "path":"cortex",
+    "path":"http://localhost:9080/cortex",
```
Comment (shaunmaharaj) [inline src/ep.config.json]: This shouldn't be pushed

File: src/components/productrecommendations.main.jsx [hunk 4]
```diff
@@ -38,27 +38,113 @@ class ProductRecommendationsDisplayMain extends React.Component {
...
+                            <div className="carousel-inner">
+                            {this.renderRecommendations()}
+                            </div>
+                        </div> 
+                        <a className="left carousel-control" href="#theCarousel" data-slide="prev"><i className="glyphicon glyphicon-chevron-left"></i></a>
+                        <a className="right carousel-control" href="#theCarousel" data-slide="next"><i className="glyphicon glyphicon-chevron-right"></i></a>                           
+                        </div>
+                    </div>
+
+                    {/* Copy this section to add new carousel */}
```
Comment (shaunmaharaj) [inline src/components/productrecommendations.main.jsx]: Is this your comment?

File: src/components/productrecommendations.main.jsx [hunk 5]
```diff
@@ -38,27 +38,113 @@ class ProductRecommendationsDisplayMain extends React.Component {
...
             }
-        })
+        }
+        
+        return data;
+
     }
     render() {
         if (this.state.productData.links.length > 0) {
             console.log(this.state.productData['_recommendations'][0]['_crosssell'][0])
```
Comment (shaunmaharaj) [inline src/components/productrecommendations.main.jsx]: Was this left intentionally?

File: src/components/productrecommendations.main.jsx [hunk 6]
```diff
@@ -38,27 +38,113 @@ class ProductRecommendationsDisplayMain extends React.Component {
...
+                        </div>    
+                    );
             }
-        })
+        }
+        
+        return data;
+
     }
     render() {
```
Comment (shaunmaharaj) [inline src/components/productrecommendations.main.jsx]: What happens if a product doesn't have any recommendations?

File: src/components/productrecommendations.main.jsx [hunk 7]
```diff
@@ -38,27 +38,113 @@ class ProductRecommendationsDisplayMain extends React.Component {
...
+                            {this.renderReplacement()}
+                            </div>
+                        </div> 
+                        <a className="left carousel-control" href="#theReplacementsCarousel" data-slide="prev"><i className="glyphicon glyphicon-chevron-left"></i></a>
+                        <a className="right carousel-control" href="#theReplacementsCarousel" data-slide="next"><i className="glyphicon glyphicon-chevron-right"></i></a>                           
+                        </div>
+                    </div>
+                
+                    
+                    <label className="control-label">Product Upsell</label>
```
Comment (shaunmaharaj) [inline src/components/productrecommendations.main.jsx]: Will you push a change later to discern if the product is an upsell, cross-sell, etc and display the appropriate carousel then?

File: src/style/style.css [hunk 8]
```diff
@@ -9775,3 +9775,34 @@ ul {
   margin: 0 auto;
   margin-top: 10px;
 }
+.carousel-inner .category-item-title {
+  min-height: 0px;
+  font-size: 1.4rem;
+ }
+ .carousel .left.carousel-control +.right.carousel-control {
```
Comment (shaunmaharaj) [inline src/style/style.css]: Can you please update the arrow button to remove the gradient and add a width?
ex.
```
.carousel .left.carousel-control +.right.carousel-control {
    width: 65px;
    border-bottom-left-radius: 20px;
    border-top-left-radius: 20px;
    height: 33%;
    top: 33%;
    background-color: gray;
}
```

File: package.json (hunk 1 above)
Comment (jatinbehlEP) [reply package.json]: fixed

File: package.json (hunk 2 above)
Comment (jatinbehlEP) [reply package.json]: fixed

File: src/ep.config.json (hunk 3 above)
Comment (jatinbehlEP) [reply src/ep.config.json]: fixed

File: src/components/productrecommendations.main.jsx (hunk 4 above)
Comment (jatinbehlEP) [reply src/components/productrecommendations.main.jsx]: yes, to make it easier for external devs to customize the code.

**Pull request merged**